### PR TITLE
Feat: Boss Spawn Restrictions + necessary files for project build

### PR DESCRIPTION
**Pull request by Davoleo**
This PR's content:
- `gradlew`, `gradlew.bat`, `gradle-wrapper.properties` and `gradle-wrapper.jar`
  - These Files are necessary to build the project but Brian-Wuest ignored them by mistake I'm adding them back.
- Boss Spawn Restrictions System

**Currently supported restrictions:**
  - dimensions
  - time of day
  - biomes
  - Y Level
  - weather
  - groundRadius

_Note: I'll write some Markdown docs on using the restriction system through json once I'm done with it, in the meanwhile you can have a look at `sample_fights/all_the_restrictions.json` for a use example._

### CODE REVIEW

**Comment by democat3457** [comment]
Can you change the gradlew version to 4.9-bin

**Comment by Davoleo** [comment]
Done

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.8.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.9-bin.zip
```
Comment (democat3457) [inline gradle/wrapper/gradle-wrapper.properties:3]: this won't actually update the gradle wrapper jar, you need to run `./gradlew wrapper --gradle-version 4.9 --distribution-type bin`

only for version 2.14 can you just change the gradle-wrapper.properties

Comment (Davoleo) [reply gradle/wrapper/gradle-wrapper.properties:3]: mmh ok, I'll work on it tomorrow, I've tried updating it now, but it seems like it can't download forgegradle anymore so I'll need to fix that before pushing.
although I'm wondering why should we update the wrapper, it worked just fine before in my env

Comment (democat3457) [reply gradle/wrapper/gradle-wrapper.properties:3]: oof ok

I just want gradle 4.9 because it's the one most 1.12 mods use (and i'm pretty low on storage so I don't want another version of gradle in my caches folder)
tl;dr convenience

**Comment by Davoleo** [comment]
should be updated now, tell me if something breaks
